src/rpc/RPCHelpers.cpp

```diff
@@ -1411,16 +1411,19 @@ parseBook(boost::json::object const& request)
 
     if (isXRP(pay_currency) && !isXRP(pay_issuer))
         return Status{
-            Error::rpcINVALID_PARAMS,
+            Error::rpcSRC_ISR_MALFORMED,
             "Unneeded field 'taker_pays.issuer' for XRP currency "
             "specification."};
 
     if (!isXRP(pay_currency) && isXRP(pay_issuer))
         return Status{
-            Error::rpcINVALID_PARAMS,
+            Error::rpcSRC_ISR_MALFORMED,
             "Invalid field 'taker_pays.issuer', expected non-XRP "
             "issuer."};
 
+    if ((!isXRP(pay_currency)) && (!taker_pays.contains("issuer")))
```

Comment:
Imho, no real need for the extra set of parentheses here - i'd drop them but it's up to you.